docs/pages/versions/unversioned/sdk/in-app-purchases.md

```diff
@@ -0,0 +1,317 @@
+        for (const purchase of results) {
+            if (!purchase.acknowledged) {
+                console.log(`Successfully purchased ${purchase.productId}`);
+                // Process transaction here and unlock content
+
+                // Then when you're done...
+                finishTransactionAsync(purchase, true);
+            }
+        }
+    } else if (responseCode === ResponseCode.USER_CANCELED) {
+        console.log('User canceled');
+    } else { // Either received an error or deferred purchase (iOS only)
+        console.warn(
+            `Something went wrong with the purchase. Received response code ${responseCode} and errorCode ${errorCode}`
+        );
+    }
+});
+```
+
+### `InAppPurchases.getPurchaseHistoryAsync(refresh?: boolean)`
+
+Retrieves the user's previous purchase history.
+
+On Android, if refresh is set to `true` it will make a network request and return up to one entry per item even if that purchase is expired, canceled, or consumed. Use this if you want to sync purchases across devices or see purchases that are expired or consumed. If refresh is `false` it relies on the Play Store cache and returns the same result as `connectAsync`. An important caveat is that the return type when refresh is `true` is actually a subset of when it's `false`. This is because Android returns a `PurchaseHistoryRecord` which only contains the purchase time, purchase token, and product ID, rather than all of the attributes found in the `Purchase` type.
+
+On iOS, the refresh boolean is ignored and it returns the purchase history in the same way that `connectAsync` does. An important thing to note is that on iOS, Storekit actually creates a new transaction object every time you restore completed transactions, therefore the `purchaseTime` and `orderId` may be inaccurate if it's a restored purchase. If you need the original transaction's information you can use `originalPurchaseTime` and `originalOrderId`, but those will be 0 and an empty string respectively if it is the original transaction.
+
+#### Arguments
+
+- **refresh (_boolean_)** -- A boolean that indicates whether or not you want to make a network request to sync expired/consumed purchases and those on other devices (Android only)
+
+#### Returns
+
+A `Promise` that is resolved with a `QueryResponse` that contains an array of `Purchase` objects.
+
+#### Example
+```javascript
+const { responseCode, results } = await getPurchaseHistoryAsync();
+if (responseCode === ResponseCode.OK) {
+    results.forEach(result => {
+        // Handle purchase history
+    })
+}
+```
+
+### `InAppPurchases.purchaseItemAsync(productId: string, oldItem?: string)`
+
+Initiates the purchase flow to buy the item associated with this productId. This function is void and the result must be handled in the callback that you passed in to `setPurchaseListener`. This will display a prompt to the user which will allow them to either buy the item or cancel the purchase.
```

Comment:
```suggestion
Initiates the purchase flow to buy the item associated with this `productId`. This will display a prompt to the user which will allow them to either buy the item or cancel the purchase. When the purchase completes, the result must be handled in the callback that you passed in to `setPurchaseListener`. 
```

later:

```
#### Returns

 A `Promise` that is resolved when the purchase flow starts. To get the actual result of the purchase, you must handle purchase events with `setPurchaseListener`.
```